hibernate-core/src/main/java/org/hibernate/type/descriptor/java/ClobTypeDescriptor.java

```diff
@@ -107,6 +107,16 @@ else if (Clob.class.isAssignableFrom( type )) {
 						: value;
 				return (X) clob;
 			}
+			else if ( String.class.isAssignableFrom( type ) ) {
+				if ( ClobImplementer.class.isInstance( value ) ) {
```

Comment:
So two patterns show up above and it seems great to make them consistent. I think 'isAssignableFrom' is better than the deprecated 'Class#isInstance()', though even `instanceof` is enough for only static class is used here.

Reply:
Class#isInstance isn't deprecated, but I agree that `instanceof` is better here since we have the class available statically.

I can't replace with `isAssignableFrom` because that method accepts ` parameter of `Class<?>`, which is not the type of `value`